### PR TITLE
Add kubernetes v1.12 to e2e tests

### LINKED ISSUE #337: Add support for kubernetes v1.12
When adding Kubernetes v1.12 to the e2e tests, the tests fail.
Instances are not coming up (Kubelet is not registering at the Apiserver)

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
add kubernetes v1.12 to e2e tests

Fixes #337

### CODE REVIEW

**Comment by alvaroaleman** [comment]
@mrIncompetent PTAL, the test failures are all on Openstack which I presume not to be caused by this PR

File: test/e2e/provisioning/verify.go [hunk 1]
```diff
@@ -109,10 +108,11 @@ func createAndAssure(machineDeployment *v1alpha1.MachineDeployment,
 		return false, nil
 	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to created the new machineSet, err: %v", err)
+		return nil, fmt.Errorf("failed waiting for machineDeployment %s to get a node: %v", machineDeployment.Name, err)
 	}
+	glog.Infof("Found a node for machineDeployment %s", machineDeployment.Name)
 
-	glog.Infof("waiting for status = %s to come \n", corev1.NodeReady)
+	glog.Infof("Waiting for node of machineDeployment %s to become ready", machineDeployment.Name)
```
Comment (mrIncompetent) [inline test/e2e/provisioning/verify.go]: Kind's should be uppercase in logs: `machineDeployment` -> `MachineDeployment`

Comment (mrIncompetent) [reply test/e2e/provisioning/verify.go]: In all messages actually

File: test/e2e/provisioning/verify.go [hunk 2]
```diff
@@ -109,10 +108,11 @@ func createAndAssure(machineDeployment *v1alpha1.MachineDeployment,
 		return false, nil
 	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to created the new machineSet, err: %v", err)
+		return nil, fmt.Errorf("failed waiting for machineDeployment %s to get a node: %v", machineDeployment.Name, err)
```
Comment (mrIncompetent) [inline test/e2e/provisioning/verify.go]: https://github.com/kubermatic/machine-controller/pull/336/files#r222371440

File: test/e2e/provisioning/verify.go [hunk 3]
```diff
@@ -130,7 +130,7 @@ func createAndAssure(machineDeployment *v1alpha1.MachineDeployment,
 		return false, nil
 	})
 	if err != nil {
-		return nil, fmt.Errorf("falied to created the new machine, err = %v", err)
+		return nil, fmt.Errorf("failed waiting for machine %s to get a node in ready state: %v", machineDeployment.Name, err)
```
Comment (mrIncompetent) [inline test/e2e/provisioning/verify.go]: I know was already there before, but please: https://github.com/kubermatic/machine-controller/pull/336/files#r222371440

File: test/e2e/provisioning/verify.go [hunk 4]
```diff
@@ -176,7 +176,7 @@ func deleteAndAssure(machineDeployment *v1alpha1.MachineDeployment,
 		}
 		return true, nil
 	}); err != nil {
-		return fmt.Errorf("failed to wait for machines to be deleted, err = %v", err)
+		return fmt.Errorf("failed to wait for machines of machineDeployment %s to be deleted: %v", machineDeployment.Name, err)
```
Comment (mrIncompetent) [inline test/e2e/provisioning/verify.go]: I know was already there before, but please: https://github.com/kubermatic/machine-controller/pull/336/files#r222371440

File: pkg/cloudprovider/provider/hetzner/provider.go [hunk 5]
```diff
@@ -200,7 +200,11 @@ func (p *provider) Create(machine *v1alpha1.Machine, _ cloud.MachineUpdater, use
 		PublicKey: sshkey.PublicKey,
 	})
 	if err != nil || res.StatusCode != http.StatusCreated {
-		return nil, fmt.Errorf("creating temporary ssh key failed with http status %d and error %v", res.StatusCode, err)
+		var statusCode int
+		if res != nil {
+			statusCode = res.StatusCode
+		}
+		return nil, fmt.Errorf("creating temporary ssh key failed with http status %d and error %v", statusCode, err)
```
Comment (mrIncompetent) [inline pkg/cloudprovider/provider/hetzner/provider.go]: Return a separate error each case:
- `err != nil`
- `res.StatusCode != http.StatusCreated`

File: .circleci/config.yml [hunk 6]
```diff
@@ -77,9 +77,7 @@ jobs:
           name:  Run E2E test
           command: >-
               export KUBECONFIG=$GOPATH/src/github.com/kubermatic/machine-controller/.kubeconfig &&
-              go test -race -tags=e2e -parallel 240 -v -timeout 30m  ./test/e2e/... -identifier=$CIRCLE_BUILD_NUM ||
-              (kubectl logs -n kube-system $(kubectl get pods -n kube-system|egrep '^machine-controller'|
```
Comment (mrIncompetent) [inline .circleci/config.yml]: Should we upload the logs to a private gcp bucket?

Comment (alvaroaleman) [reply .circleci/config.yml]: Maybe? I don't find it extremely helpful tbh.

Comment (mrIncompetent) [reply .circleci/config.yml]: Then we'll add uploading to a bucket when we see the need for it

**Pull request merged**